client/my-sites/upgrades/domain-management/email-forwarding/email-forwarding-add-new.jsx

```diff
@@ -124,6 +146,10 @@ const EmailForwardingAddNew = React.createClass( {
 			return null;
 		}
 
+		const isValidMailbox = this.isValid( 'mailbox' ),
```

Comment:
You should merge this `const` declaration with the one above. I'd prefer to move the `! this.shouldShowForm()` check to the very top, so that we don't we can quit early and don't unnecessarily compute those values, but I'm not sure if it's not against our coding guidelines.